### ISSUE TITLE
Only act on a subset of a record

### ISSUE THREAD

**Issue by thufschmitt** [tweag/nickel#1693]
**Is your feature request related to a problem? Please describe.**

I frequently end-up in a situation where I have a Nickel expression evaluating to a big(ish) nested record, and would like to evaluate a specific field of it, rather than the whole record (both because the rest would be useless noise and for performance reasons).

**Describe the solution you'd like**

A flag similar to Nix's [`--attr`](https://nixos.org/manual/nix/stable/command-ref/nix-instantiate#opt-attr) to restrict the evaluation to a specific subfield.

**Describe alternatives you've considered**

This can be encoded:

I can do that with something like `nickel eval <<<'(import "./project.ncl").x.y.z'`, but it's rather verbose and error-prone for something that I end-up wanting fairly frequently.

I can also `jq` my way out of the result of `nickel export project.ncl`, but it doesn't solve the performance issue, nor is it practical

**Additional context**

Not super high prio, but would be a nice QoL improvement for me

**Comment by bew** [comment]
(random idea)
We could have a similar notation than Nix's flake attribute reference, where we could do:
```shell
nickel eval ./project.ncl#x.y.z
```
to access `(import "./project.ncl").x.y.z`

**Comment by yannham** [comment]
>  We could have a similar notation than Nix's flake attribute reference, where we could do:

I would avoid `#`, it led me to strange things being evaluated as a comment under zsh :sweat_smile: . About the more general idea, I believe it works a bit less well with new current CLI (to be published in 1.3), which can take many files and merge them:

```console
nickel eval foo.ncl bar.ncl blarg.ncl
```

Where would the annotation go in this case?

Otherwise, we had the simple idea of just having a normal argument `--path`, which already exist for `nickel query` (and would coincide with the more general one that could cover `eval`, `query`, `export`, and `doc`).

**Comment by yannham** [comment]
Fixed in 1.3.0 by the introduction of `--field` for most subcommands.
